src/main/java/com/uber/cadence/internal/worker/LocalActivityWorker.java

```diff
@@ -85,46 +85,73 @@ public void start() {
 
   @Override
   public boolean isStarted() {
+    if (poller == null) {
```

Comment:
It seems like poller is `NoopSuspendableWorker` initially (that is when it is not started). We should not need those null checks here.

Reply:
Good catch. Remove all of these and moved common methods implementations into an abstract base class.